snuba/cli/multistorage_consumer.py

```diff
@@ -0,0 +1,200 @@
+from snuba import environment, settings
+from snuba.consumer import MultistorageConsumerProcessingStrategyFactory
+from snuba.datasets.storages import StorageKey
+from snuba.datasets.storages.factory import WRITABLE_STORAGES, get_writable_storage
+from snuba.environment import setup_logging, setup_sentry
+from snuba.utils.metrics.wrapper import MetricsWrapper
+from snuba.utils.streams.backends.kafka import (
+    KafkaConsumer,
+    KafkaConsumerWithCommitLog,
+    build_kafka_consumer_configuration,
+    build_kafka_producer_configuration,
+)
+from snuba.utils.streams.processing import StreamProcessor
+from snuba.utils.streams.types import Topic
+
+
+logger = logging.getLogger(__name__)
+
+
+@click.command()
+@click.option(
+    "--storage",
+    "storage_names",
+    default=["events"],
+    type=click.Choice([storage_key.value for storage_key in WRITABLE_STORAGES.keys()]),
+    multiple=True,
+)
+@click.option(
+    "--consumer-group", default="snuba-consumers",
+)
+@click.option(
+    "--max-batch-size",
+    default=settings.DEFAULT_MAX_BATCH_SIZE,
+    type=int,
+    help="Max number of messages to batch in memory before writing to Kafka.",
+)
+@click.option(
+    "--max-batch-time-ms",
+    default=settings.DEFAULT_MAX_BATCH_TIME_MS,
+    type=int,
+    help="Max length of time to buffer messages in memory before writing to Kafka.",
+)
+@click.option(
+    "--auto-offset-reset",
+    default="error",
+    type=click.Choice(["error", "earliest", "latest"]),
+    help="Kafka consumer auto offset reset.",
+)
+@click.option(
+    "--queued-max-messages-kbytes",
+    default=settings.DEFAULT_QUEUED_MAX_MESSAGE_KBYTES,
+    type=int,
+    help="Maximum number of kilobytes per topic+partition in the local consumer queue.",
+)
+@click.option(
+    "--queued-min-messages",
+    default=settings.DEFAULT_QUEUED_MIN_MESSAGES,
+    type=int,
+    help="Minimum number of messages per topic+partition librdkafka tries to maintain in the local consumer queue.",
+)
+@click.option(
+    "--processes", type=int, default=1,
+)
+@click.option(
+    "--input-block-size", type=int, default=int(32 * 1e6),
+)
+@click.option(
+    "--output-block-size", type=int, default=int(32 * 1e6),
+)
+@click.option("--log-level")
+def multistorage_consumer(
+    storage_names: Sequence[str],
+    consumer_group: str,
+    max_batch_size: int,
+    max_batch_time_ms: int,
+    auto_offset_reset: str,
+    queued_max_messages_kbytes: int,
+    queued_min_messages: int,
+    processes: int,
+    input_block_size: int,
+    output_block_size: int,
+    log_level: Optional[str] = None,
+) -> None:
+
+    setup_logging(log_level)
+    setup_sentry()
+
+    storages = {
+        key: get_writable_storage(key)
+        for key in (getattr(StorageKey, name.upper()) for name in storage_names)
+    }
+
+    topics = {
+        storage.get_table_writer()
+        .get_stream_loader()
+        .get_default_topic_spec()
+        .topic_name
+        for storage in storages.values()
+    }
```

Comment:
Agree that it's not a great long term solution, and settings is a better fit. If we do end up providing the topic and commit log topic options on the command line as a short term solution though, why not just `--topic topic` or `--topic wals-rocko` though, since only a single shared topic is supported between all the storages?

Reply:
I think that would be fine for those two topics, but we would still need to come up for a per-storage syntax for replacements, since we don't use the default for the `events` replacements topic (and I don't think we'd want `events` and `errors` sharing the same replacement topic.)

I'll throw together the settings change as a follow-up to this and start the ball rolling on deprecating the topic options to all of the commands. That doesn't go as far as I'd like to solve the stream configuration issues long-term but is at least a step in the right direction to consolidating configuration.